tests/prague/eip7692_eof_v1/eip7069_extcall/test_returndataload.py

```diff
@@ -0,0 +1,265 @@
+"""
+abstract: Tests [EIP-7069: Revamped CALL instructions](https://eips.ethereum.org/EIPS/eip-7069)
+    Tests for the RETURNDATALOAD instriction
+"""  # noqa: E501
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+)
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.eof.v1.constants import NON_RETURNING_SECTION
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .. import EOF_FORK_NAME
+from . import REFERENCE_SPEC_GIT_PATH, REFERENCE_SPEC_VERSION
+from .helpers import slot_code_worked, value_code_worked
+
+REFERENCE_SPEC_GIT_PATH = REFERENCE_SPEC_GIT_PATH
+REFERENCE_SPEC_VERSION = REFERENCE_SPEC_VERSION
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+@pytest.mark.parametrize("container_style", ["EOF", "Legacy"], ids=lambda x: x)
+@pytest.mark.parametrize(
+    "returndata",
+    [
+        b"",
+        b"\x10" * 0x10,
+        b"\x20" * 0x20,
+        b"\x30" * 0x30,
+    ],
+    ids=lambda x: "len_%x" % len(x),
+)
+@pytest.mark.parametrize(
+    "offset",
+    [
+        0,
+        0x10,
+        0x20,
+        0x30,
+    ],
+    ids=lambda x: "offset_%x" % x,
+)
+@pytest.mark.parametrize(
+    "size",
+    [
+        0,
+        0x10,
+        0x20,
+        0x30,
+    ],
+    ids=lambda x: "size_%x" % x,
+)
+def test_returndatacopy_handling(
+    state_test: StateTestFiller,
+    container_style: str,
+    returndata: bytes,
+    offset: int,
+    size: int,
+):
+    """
+    Tests ReturnDataLoad including multiple offset conditions and differeing legacy vs. eof
+    boundary conditions.
+
+    entrypoint creates a "0xff" test area of memory, delegate calls to caller.
+    Caller is either EOF or legacy, as per parameter.  Calls returner and copies the return data
+    based on offset and size params.  Cases are expected to trigger boundary violations.
+
+    Entrypoint copies the test area to storage slots, and the expected result is asserted.
+    """
+    env = Environment()
+    address_entry_point = Address(0x1000000)
+    address_caller = Address(0x1000001)
+    address_returner = Address(0x1000002)
+    tx = Transaction(to=address_entry_point, gas_limit=2_000_000, nonce=1)
+
+    slot_result_start = 0x1000
+
+    pre = {
+        TestAddress: Account(balance=10**18, nonce=tx.nonce),
+        address_entry_point: Account(
+            nonce=1,
+            code=Op.NOOP
+            # First, create a "dirty" area, so we can check zero overwrite
+            + Op.PUSH1(1)
+            + Op.PUSH0
+            + Op.SUB
+            + Op.MSTORE(0x00, Op.DUP1)
+            + Op.MSTORE(0x20, Op.DUP1)
```

Comment:
Yes it would, that's what took most of the time planning for this PR was contemplating how I would prove the zero extension at interesting lengths.  We could also make a test or two of absurd lengths too (2k/128k)

Reply:
Sounds good, I created an issue, so I'll ping you once that is done and we can revisit these tests then.